Common/src/toolboxes/MMS/CMMSNSTwoHalfSpheresSolution.cpp

```diff
@@ -0,0 +1,402 @@
+    SU2_MPI::Error("Standard air or ideal gas must be selected for the MMS NS Two Half Spheres case",
+                   CURRENT_FUNCTION);
+
+  if(config->GetKind_ViscosityModel() != CONSTANT_VISCOSITY)
+    SU2_MPI::Error("Sutherland must be selected for viscosity for the MMS NS Two Half Spheres case",
+                   CURRENT_FUNCTION);
+
+  if(config->GetKind_ConductivityModel() != CONSTANT_PRANDTL)
+    SU2_MPI::Error("Constant Prandtl number must be selected for the MMS NS Two Half Spheres case",
+                   CURRENT_FUNCTION);
+}
+
+CMMSNSTwoHalfSpheresSolution::~CMMSNSTwoHalfSpheresSolution(void) { }
+
+void CMMSNSTwoHalfSpheresSolution::GetBCState(const su2double *val_coords,
+                                              const su2double val_t,
+                                              su2double       *val_solution) {
+
+  /*--- The exact solution is prescribed on the boundaries. ---*/
+  GetSolution(val_coords, val_t, val_solution);
+}
+
+void CMMSNSTwoHalfSpheresSolution::GetSolution(const su2double *val_coords,
+                                               const su2double val_t,
+                                               su2double       *val_solution) {
+
+  /* Easier storage of the x-, y- and z-coordinates. */
+  const su2double x = val_coords[0];
+  const su2double y = val_coords[1];
+  const su2double z = val_coords[2];
+
+  /* Determine the dimensional solution for the temperature. */
+  const su2double Pi   = PI_NUMBER;
+  const su2double r    = sqrt(x*x + y*y + z*z);
+  const su2double fact = (r-1.0)*(r-1.0)/(a_T1 + a_T2);
+
+  su2double T = 0.25*TWall*(3.0 + fact*(a_T1*cos(Pi*(r-2.0))
+              +                         a_T2*cos(Pi*(r-2.0)*2.0)));
+
+  /* Determine the dimensional solution for the velocities. */
+  su2double u = u_0*(r-1.0)*(2.0-r)*4.0;
+  su2double v = v_0*(r-1.0)*(2.0-r)*4.0;
+  su2double w = w_0*(r-1.0)*(2.0-r)*4.0;
+
+  /* Compute the pressure from the density and temperature. */
+  su2double rho = rho_0;
+  su2double p   = rho*RGas*T;
+
+  /* Determine the non-dimensional solution. */
+  rho /= Density_Ref;
+  p   /= Pressure_Ref;
+  u   /= Velocity_Ref;
+  v   /= Velocity_Ref;
+  w   /= Velocity_Ref;
+
+  /* Determine the non-dimensional conserved variables. */
+  val_solution[0] = rho;
+  val_solution[1] = rho*u;
+  val_solution[2] = rho*v;
+  val_solution[3] = rho*w;
+  val_solution[4] = p/(Gamma-1.0) + 0.5*rho*(u*u + v*v + w*w);
+}
+
+void CMMSNSTwoHalfSpheresSolution::GetMMSSourceTerm(const su2double *val_coords,
+                                                    const su2double val_t,
+                                                    su2double       *val_source) {
+
+  /*--- Abbreviate Pi and the coordinates. ---*/
+  const su2double Pi = PI_NUMBER;
+  const su2double x = val_coords[0];
+  const su2double y = val_coords[1];
+  const su2double z = val_coords[2];
+
+  /*--- The source code for the source terms is generated in Maple. ---*/
```

Comment:
If small enough, can you include the steps used to generate this kind of machine generated code?

Reply:
Yes, that is possible. However, any symbolic package can do this, and we did not want to make the impression that this has to be done with Maple or python sympy. If people think it will clarify matters, we can certainly add this information.

Reply:
I think it is clear, it is just in case someone wants to reproduce the same steps, serves as mini tutorial almost.

Reply:
Would it be an idea to include the Maple and Python scripts in a separate directory then?

Reply:
A good one I think

Reply:
Should we wait for this before we merge it in, then?

Reply:
I think adding the scripts to the VandV repo where our initial MMS cases have already been placed could be ideal. Fits in nicely, and we will eventually be able to explain everything with those examples (note that there is a working group to discuss what the docs should look like there).

Reply:
@economon, I just pushed the Maple files that generate the source code for the MMS source terms to Common/src/toolboxes/MMS/CreateMMSSourceTerms. The reason for putting in the source tree and not in the VandV repo is that it generates actual source code. However, if you prefer the VandV repo for these scripts, that is fine with me as well as long as it is properly referenced in the source code. 

BTW, I just realized that I have not added a regression test case for this feature, so do not merge yet.

Reply:
Aok @vdweide, this is also very logical and just fine for me. Since we are still working on the VandV structure anyway, let's go with posting the scripts that generate the sources directly in the folder you have created. I will also add the python scripts for the incompressible MMS cases with sympy there. Would be good to wrap this one up asap and adjust later only if necessary.

Reply:
@economon, if you add those python scripts we addressed all concerns/remarks as far as I can see it. So it is ready to be merged in.

Reply:
And after I fix an error in the regression tests.....